pms-115-120/utility/src/main/java/io/mosip/pms/ida/service/PMSDataMigrationService.java

```diff
@@ -55,46 +64,94 @@ public class PMSDataMigrationService {
 
 	@Autowired
 	MispLicenseRepository mispLicenseRepository;
+	
+	@Autowired
+	LastSyncRepository lastSyncRepository;
 
 	@Autowired
 	private Environment environment;
 
 	@Autowired
 	private ObjectMapper mapper;
+	
+	@Autowired
+	private CertUtil certUtil;
+	
+	@Autowired
+	RestUtil restUtil;
 
 	@Autowired
 	private WebSubPublisher webSubPublisher;
 
 	public void initialize() {
-		LOGGER.error("Started publishing the data");
+    	LocalDateTime lastSync = getLastSyncTimeStamp();
+    	LocalDateTime latestSync = LocalDateTime.now();
+		LOGGER.info("Started publishing the data");
 		try {
-			publishAPIKeyData();
-			publishMISPLicenseData();
+				publishPartnerUpdated(lastSync,latestSync);
+				publishAPIKeyData(lastSync,latestSync);
+				publishMISPLicenseData(lastSync,latestSync);
+				if(lastSync!=null) {
+					publishUpdateApiKey(lastSync,latestSync);
```

Comment:
why we publish API key only when lastSync is not null? In the first run we will not have any api keys ?

Reply:
for publishing new api key, we need to send certificates also along with the API key data.  For ApiKeyData update event we don't need to send any certificates sir.